src/app/custom-meals/[id]/actions.ts

```diff
@@ -0,0 +1,114 @@
+'use server'
+
+import { CartLinesAddPayload } from '@shopify/hydrogen-react/storefront-api-types'
+import { cookies } from 'next/headers'
+import { z } from 'zod'
+import { ServerActionError } from '@/@types/common'
+import storefrontApi from '@/service/api/storefrontApi'
+import {
+  SHOPIFY_CART_ID_COOKIE,
+  SHOPIFY_CUSTOMER_EMAIL_COOKIE,
+} from '@/utils/constants/cookies'
+import { StorefrontDataKey } from '@/utils/constants/storefrontGql'
+import {
+  handleStorefrontGqlResponse,
+  setCookie,
+} from '@/utils/functions/common'
+import { logger } from '@/utils/functions/logger'
+
+const AddToCartFormSchema = z.object({
+  sellingPlanId: z.string({
+    message: 'Please select a subscription option',
+  }),
+  merchandiseId: z.string({
+    message: 'Product not found',
+  }),
+})
+
+type AddToCartForm = z.infer<typeof AddToCartFormSchema>
+
+export async function addToCart(
+  _: undefined | ServerActionError<AddToCartForm>,
+  form: FormData,
+): Promise<
+  undefined | (ServerActionError<AddToCartForm> & { success?: boolean })
+> {
+  const data = {
+    sellingPlanId: form.get('sellingPlanId'),
+    merchandiseId: form.get('merchandiseId'),
+  }
+
+  const parsedData = AddToCartFormSchema.safeParse(data)
+  if (!parsedData.success) {
+    return { zodError: parsedData.error.format() }
+  }
+
+  const cookieStore = cookies()
+  const cartIdCookie = cookieStore.get(SHOPIFY_CART_ID_COOKIE)
+
+  // Add item to existing cart
+  if (cartIdCookie) {
+    const addItemToCartResponse = await storefrontApi.addItemToCart({
+      cartId: cartIdCookie.value,
+      lines: [parsedData.data],
+    })
+
+    const { error: addItemToCartError } =
+      handleStorefrontGqlResponse<CartLinesAddPayload>(
+        addItemToCartResponse,
+        StorefrontDataKey.CART_LINES_ADD,
+      )
+
+    if (addItemToCartError) {
+      logger.error(
+        `Unable to add items to cart [cartId: ${cartIdCookie.value}][merchandiseId: ${parsedData.data.merchandiseId}][sellingPlanId: ${parsedData.data.sellingPlanId}]: ${addItemToCartError}.`,
```

Comment:
Should make a 
const req = {
      cartId: cartIdCookie.value,
      lines: [parsedData.data],
    }
    
Then when you log just [req: JSON.stringify(req)]

Same comment for line 97